dhash/module.py

```diff
@@ -383,9 +466,62 @@ async def _save_hashes(self, message: nextcord.Message):
             )
             yield h
 
+    async def _get_url_hashes(self, message: nextcord.Message):
+        for url in re.findall(URL_REGEX, message.content):
+            if self.allowed_urls and not re.search(self.allowed_urls, url):
+                continue
+            try:
+                req = urllib.request.Request(url, headers=HTTP_HEADERS)
+                file = urllib.request.urlopen(req, timeout=20)
```

Comment:
If it's possible I would change that in future. However, if you insist, I can make those changes before PR, it would just take more time to fix this and test properly again.

Reply:
OK, I think I figured it out, but it needs some testing.